test/test_integration_cluster.rb

```diff
@@ -348,6 +366,32 @@ def test_fork_worker_on_refork
     refute_includes pids, get_worker_pids(1, wrkrs - 1)
   end
 
+  def test_mold_worker_on_mold_promotion
+    refork = Tempfile.new 'refork'
+    wrkrs = 3
+    cli_server "-w #{wrkrs} test/rackup/hello_with_delay.ru", config: <<~CONFIG
+      mold_worker 20
+      on_mold_promotion { File.write '#{refork.path}', 'Mold Promoted' }
+    CONFIG
+
+    pids = get_worker_pids 0, wrkrs
+
+    socks = []
+    until refork.read == 'Mold Promoted'
+      socks << fast_connect
+      sleep 0.004
+    end
+
+    100.times {
+      socks << fast_connect
+      sleep 0.004
+    }
+
+    socks.each { |s| read_body s }
+
+    refute_includes pids, get_worker_pids(1, wrkrs)
+  end
```

Comment:
Missing test case for `on_mold_shutdown`?

Reply:
doesn't seem like a particularly valuable case to run an integration test on; no existing test in here for e.g. `on_worker_shutdown` either, though I could add cases to `test/test_config.rb` for the new hooks (though not sure complete coverage there adds much value either)